x-pack/platform/packages/shared/onechat/onechat-server/src/tools.ts

```diff
@@ -46,6 +46,91 @@ export interface RegisteredTool<
   meta?: RegisteredToolMeta;
 }
 
+/**
+ * Onechat tool, as registered by built-in tool providers.
+ */
+export interface EsqlToolDefinition extends ToolDescriptor {
+  /**
+   * The ESQL Tool name.
+   */
+  name: string;
+
+  /**
+   * The ESQL query to be executed.
+   */
+  query: string;
+
+  /**
+   * Parameters that can be used in the query.
+   * Each parameter has a key identifier and metadata about its type and usage.
+   */
+  params: Record<
+    string,
+    {
+      /**
+       * The data types of the parameter. Must be one of these
+       */
+      type:
+        | 'text'
+        | 'keyword'
+        | 'long'
+        | 'integer'
+        | 'double'
+        | 'float'
+        | 'boolean'
+        | 'date'
+        | 'object'
+        | 'nested';
+
+      /**
+       * Description of the parameter's purpose or expected values.
+       */
+      description: string;
+    }
+  >;
+}
+
+export interface EsqlTool extends RegisteredTool {
+  /**
+   * The ESQL id to be executed.
+   */
+  id: string;
+
+  /**
+   * The ESQL Tool name.
+   */
+  name: string;
+
```

Comment:
(thoughts, not actionable) So, just so it's clear:

From onechat's standpoint, we don't care that `EsqlToolDefinition` extends `ToolDescriptor` or that `EsqlTool` extends `RegisteredTool`. the `ToolProvider` you register should return plain `RegisteredTool`s.

The way the provider functions under the hood, the way the tools are configured, the way they are internally executed when calling their handlers, all of this is an implementation detail of that "esql" provider.